### PR TITLE
config: Enable Musca S1

### PR DESCRIPTION
**Pull request by gbrtth**
This commit adds support to build TF-M targeting Musca S1.

### CODE REVIEW

**Comment by jainvikas8** [comment]
Please resolve the CI build issue and update the PR description.

**Comment by gbrtth** [comment]
CI build issue is resolved, and I have updated the PR description.

**Comment by jainvikas8** [comment]
Please extend the support to `test_psa_target.py` if we think its suppose to run Regression + PSA Compliance test suites (all of them)

File: build_psa_compliance.py [hunk 1]
```diff
@@ -40,6 +40,11 @@
         "tgt_dev_apis_tfm_musca_b1",
         "tgt_ff_tfm_musca_b1",
     ],
+    "ARM_MUSCA_S1": [
```
Comment (jainvikas8) [inline build_psa_compliance.py:43]: Is this target, able to run IPC or Firmware Framework test suite?  If not please handle it and raise an issue that could be addressed in the near future.

Comment (jainvikas8) [reply build_psa_compliance.py:43]: I don't see - `tgt_ff_tfm_musca_s1` in  https://github.com/ARM-software/psa-arch-tests/tree/master/api-tests/platform/targets

Comment (gbrtth) [reply build_psa_compliance.py:43]: > 
> 
> I don't see - `tgt_ff_tfm_musca_s1` in https://github.com/ARM-software/psa-arch-tests/tree/master/api-tests/platform/targets

Yes, you are right. I removed it.

Comment (jainvikas8) [reply build_psa_compliance.py:43]: When referring - https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/blob/master/build_psa_compliance.py#L310, As ARM_MUSCA_S1 doesn't support FF/IPC, it needs to be handled and the new issue raised.

File: test_psa_target.py [hunk 2]
```diff
@@ -246,7 +262,7 @@ def _get_parser():
         "--mcu",
         help="Build for the given MCU",
         required=True,
-        choices=["ARM_MUSCA_B1"],
+        choices=["ARM_MUSCA_B1", "ARM_MUSCA_S1"],
```
Comment (jainvikas8) [inline test_psa_target.py:265]: When referring - https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/blob/master/build_psa_compliance.py#L310, As ARM_MUSCA_S1 doesn't support FF/IPC, So please look into it. As it intends to run a binary which is not compiled.

File: test_psa_target.py [hunk 3]
```diff
@@ -152,6 +152,22 @@ def _erase_flash_storage(args, suite):
             "-Intel",
         ]
 
+    if args.mcu == "ARM_MUSCA_S1":
```
Comment (jainvikas8) [inline test_psa_target.py:155]: Please add regex test logs for auto comparison.

File: test_psa_target.py [hunk 4]
```diff
@@ -152,6 +152,22 @@ def _erase_flash_storage(args, suite):
...
+            "0xA000000",
+            "-fill",
+            "0xFF",
+            "0xA1E9000",
+            "0xA1ED000",
+            "-o",
+            "mbed-os-tf-m-regression-tests-reset-flash.hex",
+            "-Intel",
+        ]
+
```
Comment (jainvikas8) [inline test_psa_target.py:170]: Please make this as a separate commit when adding support in `test_psa_target.py`

Comment (gbrtth) [reply test_psa_target.py:170]: I have separated this change to another commit.

File: test_psa_target.py (hunk 3 above)
Comment (gbrtth) [reply test_psa_target.py:155]: Logs are added.

File: build_psa_compliance.py (hunk 1 above)
Comment (gbrtth) [reply build_psa_compliance.py:43]: Issue [#49](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/issues/49) is raised, IPC is handled below.

File: test_psa_target.py (hunk 2 above)
Comment (gbrtth) [reply test_psa_target.py:265]: Issue is handled, `build_psa_compliance.py` should exit, when running with Musca-S1 and IPC.

**Comment by jainvikas8** [comment]
@gbrtth  Please rebase before the review. Also please mention if the testing of `ARM_MUSCA_S1` is done with this `test_psa_target.py`?

File: test/logs/ARM_MUSCA_S1/CRYPTO.log [hunk 5]
```diff
@@ -0,0 +1,6 @@
+Crypto Suite Report
```
Comment (jainvikas8) [inline test/logs/ARM_MUSCA_S1/CRYPTO.log:1]: Logs added to `test/logs/ARM_MUSCA_S1/*.logs` , Are they in sync with (TF-M V 1.1)
**TF-M (Secure) + TF-M (Non-Secure) == TF-M (Secure) + Mbed OS (Non-secure)** passed results?

Also, I don't see any changes done to https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/blob/master/tfm_ns_import.yaml, some files are required to be imported here, please refer to `ARM_MUSCA_B1` example.

**Comment by gbrtth** [comment]
> 
> 
> @gbrtth Please rebase before the review. Also please mention if the testing of `ARM_MUSCA_S1` is done with this `test_psa_target.py`?

I was using the vagrant environment, but could not solve the port forwarding issue. So I built and ran the tests separately and `test_psa_target.py` without the board.

File: test/logs/ARM_MUSCA_S1/CRYPTO.log (hunk 5 above)
Comment (gbrtth) [reply test/logs/ARM_MUSCA_S1/CRYPTO.log:1]: They are in sync with TF-Mv1.1 with the exception of the CRYPTO suite. I am using https://github.com/ARMmbed/trusted-firmware-m/pull/7 , and it contains crypto related changes which are not part of TF-Mv1.1.

I am using the hardware crypto accelerator and some of the tests fails, because MD5 is not supported. But is it a known issue and documented.

File: test/logs/ARM_MUSCA_S1/CRYPTO.log [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+TOTAL TESTS     : [0-9]+
+TOTAL PASSED    : [0-9]+
+TOTAL SIM ERROR : 0
+TOTAL FAILED    : 27
+TOTAL SKIPPED   : 1
```
Comment (jainvikas8) [inline test/logs/ARM_MUSCA_S1/CRYPTO.log:6]: [Musca_S1_GNUARM_CRYPTO.log](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/files/5224056/GNUARM_CRYPTO.log) refers that there are no skipped tests, is there a reason one test skip is to be considered here?

File: test/logs/ARM_MUSCA_S1/STORAGE.log [hunk 7]
```diff
@@ -0,0 +1,6 @@
+Storage Suite Report
+TOTAL TESTS     : [0-9]+
+TOTAL PASSED    : [0-9]+
+TOTAL SIM ERROR : 0
+TOTAL FAILED    : [0-1]
```
Comment (jainvikas8) [inline test/logs/ARM_MUSCA_S1/STORAGE.log:5]: [Musca_S1_GNUARM_STORAGE.log](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/files/5224056/GNUARM_STORAGE.log) refers that there are no failed tests, is there a reason one test failed is to be considered here?

**Review by jainvikas8 — COMMENTED** [review]
> > @gbrtth Please rebase before the review. Also please mention if the testing of `ARM_MUSCA_S1` is done with this `test_psa_target.py`?
> 
> I was using the vagrant environment, but could not solve the port forwarding issue. So I built and ran the tests separately and `test_psa_target.py` without the board.

The idea of using `test_psa_target.py` is to build and run the tests on the local machine, therefore it would be easy to replicate when the CI is in place. I don't think you can use vagrant set up to use the USB target attached to the host, @Patater?

File: test_psa_target.py (hunk 2 above)
Comment (jainvikas8) [inline test_psa_target.py:265]: Please use `_get_tfm_regression_targets()` https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/blob/master/build_tfm.py#L104 by defining in `psa_builder.py` in its own commit. So that the targets are shared across all the scripts.

File: test/logs/ARM_MUSCA_S1/CRYPTO.log (hunk 6 above)
Comment (gbrtth) [reply test/logs/ARM_MUSCA_S1/CRYPTO.log]: This must have been a typo, it is changed.

File: test/logs/ARM_MUSCA_S1/STORAGE.log (hunk 7 above)
Comment (gbrtth) [reply test/logs/ARM_MUSCA_S1/STORAGE.log]: This must have been a typo, it is changed.

File: test_psa_target.py (hunk 2 above)
Comment (gbrtth) [reply test_psa_target.py]: I defined `_get_tfm_regression_targets()` in `psa_builder.py`, although it also worked being defined in `build_tfm.py`.

File: psa_builder.py [hunk 8]
```diff
@@ -77,6 +85,8 @@
 TF_M_BUILD_DIR = join(mbed_path, TF_M_RELATIVE_PATH, "TARGET_IGNORE")
 POPEN_INSTANCE = None
 
+from tools.toolchains import TOOLCHAIN_CLASSES, TOOLCHAIN_PATHS
```
Comment (jainvikas8) [inline psa_builder.py:88]: Do we need to import these?

File: test_psa_target.py (hunk 2 above)
Comment (jainvikas8) [reply test_psa_target.py]: I think we should keep in `psa_builder.py` as this is more generic.

**Comment by jainvikas8** [comment]
Please fix CI - `Black` formatting issues.

**Comment by jainvikas8** [comment]
Regarding commits, please remove change ID and add a suitable description to the commit message
The commit - https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/pull/42/commits/fd6522a0da861e5227805fc0b2f71b1baf5797dc , please move `test_psa_target.py` changes to https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/pull/42/commits/a00c2744292c98e5b9ef07c8944d040f513a7b49 as it is related to adding support to that script.

File: psa_builder.py (hunk 8 above)
Comment (gbrtth) [reply psa_builder.py]: No, I have removed that line.

**Comment by gbrtth** [comment]
> 
> 
> Regarding commits, please remove change ID and add a suitable description to the commit message

Change ID is removed. I have updated the commit messages to be more meaningful.

> The commit - [fd6522a](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/commit/fd6522a0da861e5227805fc0b2f71b1baf5797dc) , please move `test_psa_target.py` changes to [a00c274](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/commit/a00c2744292c98e5b9ef07c8944d040f513a7b49) as it is related to adding support to that script.

Test run related changes are now in one commit.

File: test_psa_target.py [hunk 9]
```diff
@@ -168,6 +184,16 @@ def _execute_test(args, suite):
...
+    # Issue : https://github.com/ARMmbed/mbed-os-tf-m-regression-tests/issues/49
+    # There is no support for this target to run Firmware Framework tests
+    if suite == "IPC" and args.mcu == "ARM_MUSCA_S1":
+        logging.info(
+            "%s config is not supported for %s target" % (suite, args.mcu)
+        )
+        test_results[suite] = "SKIPPED"
+        return
+
```
Comment (jainvikas8) [inline test_psa_target.py:196]: We should avoid building the target too with IPC config. I think it affects as good as execution

Comment (gbrtth) [reply test_psa_target.py]: I have moved this check to `_run_compliance_test` , so it happens before the build.

**Comment by jainvikas8** [comment]
@gbrtth  Thanks for addressing the comments.
As we don't have CI in place at the moment to test the target, were you able to run these from the host machine to target? 
For example:  `python3 test_target.py -t GNUARM -m ARM_MUSCA_B1 -d /media/<user>/<target> -p /dev/ttyACM0:115200`

**Review by jainvikas8 — APPROVED** [review]

**Comment by Patater** [comment]
@gbrtth Thanks very much!
